src/auth/externalAccountAuthorizedUserClient.ts

```diff
@@ -0,0 +1,335 @@
+   * @return A promise that resolves with the token refresh response containing
+   *   the requested access token and its expiration time.
+   */
+  async refreshToken(
+    refreshToken: string,
+    additionalHeaders?: Headers
+  ): Promise<TokenRefreshResponse> {
+    const values = new URLSearchParams({
+      grant_type: 'refresh_token',
+      refresh_token: refreshToken,
+    });
+
+    const headers = {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    };
+    // Inject additional headers if available.
+    Object.assign(headers, additionalHeaders || {});
+
+    const opts: GaxiosOptions = {
+      url: this.url,
+      method: 'POST',
+      headers,
+      data: values.toString(),
+      responseType: 'json',
+    };
+    // Apply OAuth client authentication.
+    this.applyClientAuthenticationOptions(opts);
+
+    try {
+      const response = await this.transporter.request<TokenRefreshResponse>(
+        opts
+      );
+      // Successful response.
+      const tokenRefreshResponse = response.data;
+      tokenRefreshResponse.res = response;
+      return tokenRefreshResponse;
+    } catch (error) {
+      // Translate error to OAuthError.
+      if (error instanceof GaxiosError && error.response) {
+        throw getErrorFromOAuthErrorResponse(
+          error.response.data as OAuthErrorResponse,
+          // Preserve other fields from the original error.
+          error
+        );
+      }
+      // Request could fail before the server responds.
+      throw error;
+    }
+  }
+}
+
+/**
+ * External Account Authorized User Client. This is used for OAuth2 credentials
+ * sourced using external identities through Workforce Identity Federation.
+ * Obtaining the initial access and refresh token can be done through the
+ * Google Cloud CLI.
+ */
+export class ExternalAccountAuthorizedUserClient extends AuthClient {
+  private cachedAccessToken: CredentialsWithResponse | null;
+  private readonly externalAccountAuthorizedUserHandler: ExternalAccountAuthorizedUserHandler;
+  private refreshToken: string;
+
+  /**
+   * Instantiates an ExternalAccountAuthorizedUserClient instances using the
+   * provided JSON object loaded from a credentials files.
+   * An error is throws if the credential is not valid.
+   * @param options The external account authorized user option object typically
+   *   from the external accoutn authorized user JSON credential file.
+   * @param additionalOptions Optional additional behavior customization
+   *   options. These currently customize expiration threshold time and
+   *   whether to retry on 401/403 API request errors.
+   */
+  constructor(
+    options: ExternalAccountAuthorizedUserClientOptions,
+    additionalOptions?: RefreshOptions
+  ) {
+    super();
+    if (options.type !== EXTERNAL_ACCOUNT_AUTHORIZED_USER_TYPE) {
+      throw new Error(
+        `Expected "${EXTERNAL_ACCOUNT_AUTHORIZED_USER_TYPE}" type but ` +
+          `received "${options.type}"`
+      );
+    }
```

Comment:
If the type (`ExternalAccountAuthorizedUserClientOptions`) has been updated to reflect `type` can only be `EXTERNAL_ACCOUNT_AUTHORIZED_USER_TYPE`, then this should never result to true.